src/main/java/com/avispl/symphony/dal/avdevices/bridges/vaddio/avbridgenano/VaddioBridgeNanoCommunicator.java

```diff
@@ -0,0 +1,1045 @@
+import java.util.stream.Collectors;
+
+import org.springframework.util.CollectionUtils;
+
+import javax.security.auth.login.FailedLoginException;
+
+import com.avispl.symphony.api.dal.control.Controller;
+import com.avispl.symphony.api.dal.dto.control.AdvancedControllableProperty;
+import com.avispl.symphony.api.dal.dto.control.ControllableProperty;
+import com.avispl.symphony.api.dal.dto.monitor.ExtendedStatistics;
+import com.avispl.symphony.api.dal.dto.monitor.Statistics;
+import com.avispl.symphony.api.dal.monitor.Monitorable;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.AudioCrosspoint;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.AudioInput;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.EnumTypeHandler;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.NetworkInformation;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.PropertiesControlList;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.StreamSettings;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.VaddioCommand;
+import com.avispl.symphony.dal.avdevices.bridges.vaddio.avbridgenano.common.VaddioNanoConstant;
+import com.avispl.symphony.dal.communicator.SshCommunicator;
+import com.avispl.symphony.dal.util.StringUtils;
+
+/**
+ * VaddioBridgeNanoCommunicator An implementation of SshCommunicator to provide communication and interaction with Vaddio Bridge Nano device
+ *
+ * Monitoring
+ * Network information
+ * Streaming Settings
+ *
+ * Controlling
+ * Mute
+ * Volume(dB)
+ * Gain(dB)
+ * Reboot
+ * Video Mute
+ * Audio Mute
+ *
+ * @author Kevin / Symphony Dev Team<br>
+ * Created on 8/14/2023
+ * @since 1.0.0
+ */
+public class VaddioBridgeNanoCommunicator extends SshCommunicator implements Monitorable, Controller {
+
+	/**
+	 * cache to store key and value
+	 */
+	private final Map<String, String> cacheKeyAndValue = new HashMap<>();
+
+	/**
+	 * count the failed command
+	 */
+	private final Map<String, String> failedMonitor = new HashMap<>();
+
+	/**
+	 * Prevent case where {@link VaddioBridgeNanoCommunicator#controlProperty(ControllableProperty)} slow down -
+	 * the getMultipleStatistics interval if it's fail to doGet the cmd
+	 */
+	private static final int controlSSHTimeout = 3000;
+	/**
+	 * Set back to default timeout value in {@link SshCommunicator}
+	 */
+	private static final int statisticsSSHTimeout = 30000;
+
+	/**
+	 * ReentrantLock to prevent telnet session is closed when adapter is retrieving statistics from the device.
+	 */
+	private final ReentrantLock reentrantLock = new ReentrantLock();
+
+	/**
+	 * Store previous/current ExtendedStatistics
+	 */
+	private ExtendedStatistics localExtendedStatistics;
+
+	/**
+	 * configManagement imported from the user interface
+	 */
+	private String configManagement;
+
+	/**
+	 * configManagement in boolean value
```

Comment:
can you please give a bit more description for the fields that have comment block to them?
like what's it used for, briefly